mlrun/api/crud/model_monitoring/_model_endpoint_stores/sql_model_endpoint_store.py

```diff
@@ -0,0 +1,551 @@
+                logger.warning(
+                    "Labels should be from type dictionary, not list,"
+                    "Will be deprecated in 1.4.0.",
+                    labels=labels,
+                )
+
+            # Convert the results from the DB into a ModelEndpoint object and append it to the ModelEndpointList
+            for endpoint_values in query.all():
+                endpoint_dict = dict(zip(columns, endpoint_values))
+
+                # Filter labels
+                if labels and labels != json.loads(
+                    endpoint_dict.get(model_monitoring_constants.EventFieldType.LABELS)
+                ):
+                    continue
+                endpoint_obj = self._convert_into_model_endpoint_object(endpoint_dict)
+
+                # If time metrics were provided, retrieve the results from the time series DB
+                if metrics:
+                    if endpoint_obj.status.metrics is None:
+                        endpoint_obj.status.metrics = {}
+                    endpoint_metrics = self.get_endpoint_real_time_metrics(
+                        endpoint_id=endpoint_obj.metadata.uid,
+                        start=start,
+                        end=end,
+                        metrics=metrics,
+                    )
+                    if endpoint_metrics:
+
+                        endpoint_obj.status.metrics[
+                            model_monitoring_constants.EventKeyMetrics.REAL_TIME
+                        ] = endpoint_metrics
+
+                endpoint_list.endpoints.append(endpoint_obj)
+            session.close()
+        return endpoint_list
+
+    @staticmethod
+    def _filter_values(
+        query: db.orm.query.Query,
+        model_endpoints_table: db.Table,
+        key_filter: str,
+        filtered_values: typing.List,
+        combined=True,
+    ) -> db.orm.query.Query:
+        """Filtering the SQL query object according to the provided filters.
+
+        :param query:                 SQLAlchemy ORM query object. Includes the SELECT statements generated by the ORM
+                                      for getting the model endpoint data from the SQL table.
+        :param model_endpoints_table: SQLAlchemy table object that represents the model endpoints table.
+        :param key_filter:            Key column to filter by.
+        :param filtered_values:       List of values to filter the query the result.
+        :param combined:              If true, then apply AND operator on the filtered values list. Otherwise, apply OR
+                                      operator.
+
+        return:                      SQLAlchemy ORM query object that represents the updated query with the provided
+                                     filters.
+        """
+
+        # Generating a tuple with the relevant filters
+        filter_query = ()
+        for _filter in filtered_values:
+            filter_query += (model_endpoints_table.c[key_filter] == _filter,)
```

Comment:
Note that logically if you have multiple values and you set `combined = True` then you're guaranteed to not get any results (because a value cannot be A and B). Suggest you throw an exception there, as a sanity check.
Also, if `combined = False` it's much better to replace the long `OR` concat with a single `query.filter(model_endpoints_table.c[key_filter].in_(filtered_values))`.